test/integration_test/migration_failover_failback_test.go

```diff
@@ -0,0 +1,174 @@
+		// check if the app is scaled down.
+		updatedScaleFactor, err := schedulerDriver.GetScaleFactorMap(ctxs[0])
+		if err != nil {
+			return "", true, err
+		}
+
+		for k := range updatedScaleFactor {
+			if int(updatedScaleFactor[k]) != 0 {
+				return "", true, fmt.Errorf("expected scale to be 0")
+			}
+		}
+		return "", false, nil
+	}
+	fmt.Println("starting task to wait on scale down")
+	_, err = task.DoRetryWithTimeout(tk, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Unexpected error on scaling down application.")
+
+	// start the app on cluster 2
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting remote config")
+
+	fmt.Println("scaling application $$$: ", oldScaleFactor)
+	// Set scale factor to it's orignal values on cluster 2
+	err = schedulerDriver.ScaleApplication(preMigrationCtx, oldScaleFactor)
+	require.NoError(t, err, "Unexpected error on ScaleApplication")
+
+	err = schedulerDriver.WaitForRunning(preMigrationCtx, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state on remote cluster after migration")
+}
+
+func testMigrationFailback(
+	t *testing.T,
+	preMigrationCtx *scheduler.Context,
+	ctxs []*scheduler.Context,
+) {
+	// Failback the application
+	// Trigger a reverse migration
+
+	ctxsReverse, err := schedulerDriver.Schedule("mysql-migration-failover-failback",
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-enc-pvc"}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxsReverse), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxsReverse[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for app to get to running state")
+
+	postMigrationCtx := ctxsReverse[0].DeepCopy()
+
+	// create, apply and validate cluster pair specs
+	err = scheduleClusterPair(ctxsReverse[0], false, false, "cluster-pair-reverse", true)
+	require.NoError(t, err, "Error scheduling cluster pair")
+
+	// apply migration specs
+	err = schedulerDriver.AddTasks(ctxsReverse[0],
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-migration-failover-failback"}})
+	require.NoError(t, err, "Error scheduling migration specs")
+
+	err = schedulerDriver.WaitForRunning(ctxsReverse[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for migration to complete")
+
+	// destroy the app on cluster 2
+	err = schedulerDriver.Destroy(preMigrationCtx, nil)
+	require.NoError(t, err, "Error destroying ctx: %+v", preMigrationCtx)
+	err = schedulerDriver.WaitForDestroy(preMigrationCtx, defaultWaitTimeout)
+	require.NoError(t, err, "Error waiting for destroy of ctx: %+v", preMigrationCtx)
+
+	// ensure app starts on cluster 1
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "Error resetting remote config")
+
+	err = schedulerDriver.WaitForRunning(postMigrationCtx, defaultWaitTimeout, defaultWaitInterval)
```

Comment:
you will need to start app on source cluster , since `appstart` in migration is set to false